.github/workflows/check-compatibility.yml

```diff
@@ -0,0 +1,35 @@
+---
+name: Check Compatibility
+
+on:
+  pull_request_target
```

Comment:
This will run on code that's checked in, not on the PR, won't it?

Reply:
I believe we want to run on each PR. This will just list the components incompatible or compatible with the change but won't block the PR in anyway unless the gradle task itself fails.

Reply:
@dblock You are right, this will run the check against whatever is in main, not the incoming pull request.

Note; This what you want from a security perspective otherwise someone could alter the job when making a pull request and have it do Bad Things (tm) to the repository

Reply:
Created an issue to track this [1]

- [1] https://github.com/opensearch-project/OpenSearch/issues/9371

Reply:
Created pull request to fix this https://github.com/opensearch-project/OpenSearch/pull/9374
Regarding security perspective, we do not integrate GH token with anything related to gradle action. So unless a change happens in core repo workflow, we should be good. https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/check-compatibility.yml#L25-L38